enhancements/ingress/add-headers-to-haproxyconfig-via-ingress-controller-or-route-api.md

```diff
@@ -0,0 +1,295 @@
+---
+title: add-headers-to-haproxyconfig-via-ingress-controller-or-route-api
+authors:
+  - "@miheer"
+reviewers:
+  - "@frobware"
+  - "@gcs278"
+  - "@Miciah"
+approvers:
+  - "@frobware"
+  - "@gcs278"
+  - "@Miciah"
+api-approvers:
+  - "@frobware"
+  - "@gcs278"
+  - "@Miciah"
+creation-date: 2020-06-10
+last-updated: 2020-06-10
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-982"
+status: implemented
+see-also:
+- ""
+replaces:
+superseded-by:
+---
+
+# Set HTTP Headers to haproxy.config either via Ingress Contoller or Route API
+
+## Summary
+
+This enhancement extends the IngressController logging API and router API to allow the user to set 
+capture of HTTP headers in the haproxy.config file. Using the new API, users may specify a list of 
+HTTP headers either via IngressController API or Route API that should be set in the haproxy.config file.
+
+## Motivation
+
+Certain headers may need to be set for compliance reasons or other reasons. In OpenShift 3, cluster administrators could
+use a custom HAProxy configuration template to set headers, but using a custom template is not permitted in OpenShift 4.
+
+### User Stories
+
+As a cluster administrator, I need to configure an IngressController to set certain HTTP headers,
+in order to satisfy compliance requirements.
+To satisfy this use-case, the cluster administrator can either set the IngressController's or Router's spec.httpSetHeaders
+field to set the required headers.
+
+### Goals
+
+1. Enable the cluster administrator to specify HTTP headers in the IngressController CR Spec that should get set in the global section
+   of the haproxy.config.
```

Comment:
I don't see how the headers can be set in the global section of the HAProxy config. [http-request](https://www.haproxy.com/documentation/hapee/2-4r1/onepage/#4-http-request) and [http-response](https://www.haproxy.com/documentation/hapee/2-4r1/onepage/#4-http-response) can only be set in frontend/listen/backend sections.

Reply:
Fixed, the implementation details mention frontend section.

Reply:
> > So, that is the reason we don't need add-header.
> 
> I don't understand. It looks like you just described what "set-header" and "add-header" do, but I don't see any explanation of why we don't need "add-header".



> > So, that is the reason we don't need add-header.
> 
> I don't understand. It looks like you just described what "set-header" and "add-header" do, but I don't see any explanation of why we don't need "add-header".

fixed it